File: app/lua/test/tGetKey.lua

print('test.tGetKey');

tGetKey = {
    --
};

function tGetKey.testGetKey()
    local testKeyPath = './../keyStore.json';
    local testKey = _commonUtil.readBinaryFile(testKeyPath);
    print("testKey : ", testKey);

    --
    local testKeyJson = json.parse(testKey);
    -- _commonUtil.prtTable(testKeyJson['edPubkeyPem']);
    --
    local ed_pubkey = ed25519GetPubkeyNoFile(testKeyJson['edPubkeyPem']);
    print("ed_pubkey : ", ed_pubkey);
end

function tGetKey.testCase()
    --
    tGetKey.testGetKey();
end

File: app/nodejs/test/tGetKey.js
//
const fs = require('fs');

//
const cryptoSsl = require("./../linux/build/Release/crypto-ssl");

module.exports.testGetKey = async () => {
    //
    let testKeyPath = './../keyStore.json';
    let testKey = fs.readFileSync(testKeyPath, 'binary');
    console.log('testKey : ' + testKey);

    //
    let testKeyJson = JSON.parse(testKey);


    //
    console.log('edPubkeyPem : ' + testKeyJson.edPubkeyPem);
    //
    let ed_pubkey = cryptoSsl.ed25519GetPubkeyNoFile(testKeyJson.edPubkeyPem);
    console.log('ed_pubkey : ' + ed_pubkey);
}

//
module.exports.testCase = async () => {
    this.testGetKey();
}